staging/src/k8s.io/apiextensions-apiserver/pkg/apiserver/validation/validation.go

```diff
@@ -79,6 +79,34 @@ func ValidateCustomResource(fldPath *field.Path, customResource interface{}, val
 				}
 				allErrs = append(allErrs, field.NotSupported(errPath, err.Value, values))
 
+			case openapierrors.TooLongFailCode:
+				value := interface{}("")
+				if err.Value != nil {
+					value = err.Value
+				}
+				allErrs = append(allErrs, field.TooLongFail(errPath, value, err.Error()))
+
+			case openapierrors.MaxItemsFailCode:
+				value := interface{}("")
+				if err.Value != nil {
+					value = err.Value
+				}
+				allErrs = append(allErrs, field.MaxItemsFail(errPath, value, err.Error()))
+
+			case openapierrors.TooManyPropertiesCode:
+				value := interface{}("")
+				if err.Value != nil {
+					value = err.Value
+				}
+				allErrs = append(allErrs, field.TooManyProperties(errPath, value, err.Error()))
```

Comment:
these aren't exercised by unit tests... can they be?

> `go test k8s.io/apiextensions-apiserver/pkg/apiserver/validation -coverprofile=coverage.out && go tool cover -html=coverage.out`

<img src="https://user-images.githubusercontent.com/980082/159959371-386c7375-9310-40c1-bffa-98ed8d3c1562.png" width="600">

Reply:
I am not sure how to trigger those errs in unit test since the maxLength/maxItems/maxProperties are set in CRD.

Reply:
Thank you for demonstrating the way to test! I have cherry-pick with necessary fix in this PR.